openvdb/points/PointAttribute.h

```diff
@@ -531,6 +532,45 @@ inline void compactAttributes(PointDataTreeT& tree)
 ////////////////////////////////////////
 
 
+template <typename PointDataTreeT>
+OPENVDB_DEPRECATED inline void
+appendAttribute(PointDataTreeT& tree,
+                const Name& name,
+                const NamePair& type,
+                const Index strideOrTotalSize,
+                const bool constantStride,
+                Metadata::Ptr metaDefaultValue,
+                const bool hidden = false,
+                const bool transient = false)
+{
+    // default metadata value must now be provided as a raw pointer
+    appendAttribute(tree, name, type, strideOrTotalSize, constantStride,
+        metaDefaultValue.get(), hidden, transient);
+}
+
+
+template <typename ValueType,
+          typename CodecType = NullCodec,
+          typename PointDataTreeT = PointDataTree>
+OPENVDB_DEPRECATED inline void
+appendAttribute(PointDataTreeT& tree,
+                const std::string& name,
+                const ValueType& uniformValue,
+                const Index strideOrTotalSize,
+                const bool constantStride,
+                Metadata::Ptr metaDefaultValue,
+                const bool hidden = false,
+                const bool transient = false)
+{
+    // default metadata value must now be provided as a raw pointer
+    appendAttribute<ValueType, CodecType>(tree, name, uniformValue, strideOrTotalSize, constantStride,
+        metaDefaultValue.get(), hidden, transient);
```

Comment:
Does this work? Do you now not get compiler errors without an additional dynamic cast on the raw pointer here?

Reply:
Good catch, we obviously don't unit test the deprecated methods. Fixed this now.